ledger/narwhal/subdag/src/lib.rs

```diff
@@ -146,6 +146,11 @@ impl<N: Network> Subdag<N> {
         // Ensure the leader certificate is an even round.
         Ok(Self { subdag })
     }
+
+    /// Initializes a new subdag.
+    pub fn from_unchecked(subdag: BTreeMap<u64, IndexSet<BatchCertificate<N>>>) -> Self {
```

Comment:
nit: a more `std`-like name for such a function would be `from_unchecked_parts` or `from_raw_parts`; I'd personally prefer it, as it clearly indicates that the underlying objects are of interest, as opposed to an alternative representation of the same object

the same suggestion applies to the other instances of this name